cloud/deploys/bin/azure/lib.sh

```diff
@@ -71,8 +71,79 @@ function azure::add_generated_secrets() {
   for key in "$@";
   do
     echo "Generating secret: ${key}"
-    local SECRET_VALUE="$(head /dev/urandom | LC_CTYPE=C tr -dc "${CHARSET}" | cut -c -40)"
+    local SECRET_VALUE="$(head /dev/urandom | LC_CTYPE=C tr -dc "${CHARSET}" |LC_CTYPE=C cut -c -40)"
     echo "Setting secret: ${key}"
     azure::add_secret "${VAULT_NAME}" "${key}" "${SECRET_VALUE}"
   done
 }
+
+#######################################
+# Get a secret value from the key vault
+# Arguments:
+#   1: The name of the key vault
+#   2: The name of the secret (used to identify it e.g. "postgres-password")
+#######################################
+function azure::get_secret_value() {
+  az keyvault secret show --vault-name "${1}" --name "${2}" --query value -o tsv
+}
+
+#######################################
+# Assign the role 'Storage Blob Data Contributor' to the current user
+#######################################
+function azure::assign_storage_blob_data_contributor_role_to_user() {
+  local user_id=$(az ad signed-in-user show --query objectId -o tsv)
+  local subscription_id=$(az account show --query id -o tsv)
+  local role_assignments=$(az role assignment list --assignee ${user_id})
+  if [[ ("Storage Blob Data Contributor" == *$role_assignments*) ]];
+  then 
+    echo "Current user already has Storage Blob Data Contributor role"
+  else
+    az role assignment create \
+      --role "ba92f5b4-2d11-453d-a403-e96b0029c9fe" \
+      --scope "subscriptions/${subscription_id}/resourceGroups/${resource_group}" \
+      --assignee-object-id "${user_id}" \
+      --assignee-principal-type "User"
+  fi
+}
+
+#######################################
+# Create storage account
+# Arguments:
+#   1: The resource group name for the storage account
+#   2: The region (e.g. EastUS) to create the storage account in
+#   3: The name of the storage account
+#######################################
+function azure::create_storage_account {
+  az storage account create \
+    --resource-group "${1}" \
+    --location "${2}" \
+    --name "${3}" \
+    --allow-blob-public-access false \
+    --min-tls-version "TLS1_2"
+}
```

Comment:
issue (blocking): it looks like there may be some duplication between the functions in this file and the ones in `cloud/azure/bin/lib/azure_log.sh`